docs/BindingWithGrammar.md

```diff
@@ -0,0 +1,131 @@
+```dtd
+<!ELEMENT foo (bar)>
+<!ELEMENT bar EMPTY>
+```
+
+Either file can be bound using the [XML Binding Wizard](#the-xml-binding-wizard), which can be triggered by [Command](#command), [CodeLens](#codelens) or [Quick Fix](#quick-fix).
+
+### The XML Binding Wizard
+
+After opening the binding wizard using one of the methods mentioned above, a dropdown with the following 2 options will appear at the Command Palette:
+
+![Binding Wizard Dropdown](./images/BindingWithGrammar/BindingWizardDropdown.png)
+
+1. Standard (xsi, DOCTYPE)
+
+If the selected file is an XSD file, this option will bind the grammar file by adding the `xmlns:xsi` (with the value as "http://www.w3.org/2001/XMLSchema-instance") and `xsi:noNamespaceSchemaLocation` (with the value as the path to the grammar file) attributes to the root tag:
+
+```xml
+<foo xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:noNamespaceSchemaLocation="foo.xsd">
+    <bar />
+</foo>
+```
+
+If the selected file is a DTD file, this option will bind the grammar file by adding a `<!DOCTYPE <root-tag-name> SYSTEM "<grammar-file-path>">` tag at the beginning of the document:
+
+```xml
+<!DOCTYPE foo SYSTEM "foo.dtd">
+<foo>
+    <bar />
+</foo>
+```
+
+2. XML Model association
+
+This option will bind the grammar file by adding the `xml-model` tag at the beginning of the document, with an `href` value as the path to the grammar file:
+
+```xml
+<?xml-model href="foo.xsd"?>
+<foo>
+    <bar />
+</foo>
+```
+
+The same applies when selecting `foo.dtd`.
+
+After selecting an option, you can select the desired grammar file using your file explorer.
+
+### Command
+
+The command "XML: Bind to grammar/schema file" can be executed from the VSCode command palette, as well as be bound to a shortcut.
+
+![Bind To Grammar Command](./images/BindingWithGrammar/BindToGrammarCommand.png)
+
+If the current XML document is not bound using an `xml-model` tag or a root element doesn't have a `xsi:noNameSpaceLocation` attribute, the command will have the user select a existing `.xsd/.dtd` grammar file to bind to the XML document.
+
+Otherwise, an error will be thrown on the client if the XML document already has a bound grammar/schema.
+
+### CodeLens
+
+When [CodeLenses](./Preferences.md#code-lens) are enabled, an unbound XML document will display a CodeLens above the root element as follows:
+
+![Bind To Grammar Command](./images/BindingWithGrammar/BindToGrammarCodeLens.png)
+
+### Quick Fix
+
+For any unbound XML document, a Quick Fix suggestion will appear beside the root element.
+
+![Bind To Grammar CodeAction](./images/BindingWithGrammar/BindToGrammarCodeAction.png)
+
+With the cursor on the first opening tag, use `Ctrl + .`, `Quick Fix...` or the lightbulb that appears and select the "Open binding wizard to bind existing grammar/schema"
+
+![Bind To Grammar CodeAction Dropdown](./images/BindingWithGrammar/BindToGrammarCodeActionDropdown.png)
```

Comment:
I think the image is out of dated because the quick fix title is bad.

Reply:
Could you pinpoint what part of the quick fix title is bad?

Reply:
![image](https://user-images.githubusercontent.com/1932211/127666984-76dd8887-7e0f-4e1d-b74c-7eb5845fc034.png)